failed building on Mac aarch64

While building with `PKG_CONFIG_PATH=$(brew --prefix icu4c)/lib/pkgconfig cargo install --path . -F external-harfbuzz` in cloned git repo,
has following error:

```bash
cargo:warning=In file included from layout/xetex-XeTeXFontInst.cpp:42:
  cargo:warning=layout/tectonic_xetex_layout.h:42:10: fatal error: 'harfbuzz/hb.h' file not found
  cargo:warning=   42 | #include <harfbuzz/hb.h>
  cargo:warning=      |          ^~~~~~~~~~~~~~~
  cargo:warning=1 error generated.

  --- stderr


  error occurred in cc-rs: command did not execute successfully (status code exit status: 1): env -u IPHONEOS_DEPLOYMENT_TARGET LC_ALL="C" "c++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=arm64-apple-macosx" "-mmacosx-version-min=15.4" "-I" "layout" "-I" "/Users/<user>/builds/tectonic/crates/bridge_core/support" "-I" "/opt/homebrew/Cellar/harfbuzz/11.1.0/include/harfbuzz" "-I" "/opt/homebrew/opt/freetype/include/freetype2" "-I" "/opt/homebrew/opt/libpng/include/libpng16" "-I" "/opt/homebrew/Cellar/glib/2.84.1/include/glib-2.0" "-I" "/opt/homebrew/Cellar/glib/2.84.1/lib/glib-2.0/include" "-I" "/opt/homebrew/opt/gettext/include" "-I" "/opt/homebrew/Cellar/pcre2/10.45/include" "-I" "/opt/homebrew/Cellar/graphite2/1.3.14/include" "-I" "/opt/homebrew/opt/freetype/include/freetype2" "-I" "/opt/homebrew/opt/libpng/include/libpng16" "-I" "/opt/homebrew/Cellar/graphite2/1.3.14/include" "-I" "/opt/homebrew/Cellar/icu4c@77/77.1/include" "-Wall" "-Wextra" "-Wall" "-std=c++17" "-Wall" "-Wdate-time" "-Wendif-labels" "-Wextra" "-Wformat=2" "-Wmissing-declarations" "-Wmissing-include-dirs" "-Wpointer-arith" "-Wredundant-decls" "-Wshadow" "-Wswitch-bool" "-Wundef" "-Wextra-semi" "-Wno-unused-parameter" "-Wno-implicit-fallthrough" "-fno-exceptions" "-fno-rtti" "-DXETEX_MAC=1" "-o" "/Users/<user>/builds/tectonic/target/release/build/tectonic_xetex_layout-6cf3422440bb1c93/out/54bde2384bf9f557-xetex-XeTeXFontInst.o" "-c" "layout/xetex-XeTeXFontInst.cpp"
```
When use `ls -la /opt/homebrew/Cellar/harfbuzz/11.1.0/include/harfbuzz | grep hb.h` does has a `hb.h` shows up.

System: MacOS aarch64
C/C++ tool chain: Macos original
Rust: newest stable

If build without `-F external-harfbuzz` would works fine.
Wondering the correct way to build with feautures mentioned in the docs at git HEAD?

--- a/crates/bridge_harfbuzz/Cargo.toml
+++ b/crates/bridge_harfbuzz/Cargo.toml
@@ -25,6 +25,7 @@ tectonic_bridge_graphite2 = { path = "../bridge_graphite2", version = "0.0.0-dev
 [build-dependencies]
 cc = "^1.0.66"
 tectonic_dep_support = { path = "../dep_support", version = "0.0.0-dev.0" }
+tectonic_cfg_support = { path = "../cfg_support", version = "0.0.0-dev.0" }
 
 [features]
 external-harfbuzz = []
@@ -32,3 +33,4 @@ external-harfbuzz = []
 [package.metadata.internal_dep_versions]
 tectonic_bridge_graphite2 = "2722731f9e32c6963fe8c8566a201b33672c5c5a"
 tectonic_dep_support = "5faf4205bdd3d31101b749fc32857dd746f9e5bc"
+tectonic_cfg_support = "thiscommit:aeRoo7oa"

--- a/crates/bridge_harfbuzz/build.rs
+++ b/crates/bridge_harfbuzz/build.rs
@@ -5,6 +5,8 @@
 
 #[cfg(feature = "external-harfbuzz")]
 mod inner {
+    use std::env;
+    use tectonic_cfg_support::target_cfg;
     use tectonic_dep_support::{Configuration, Dependency, Spec};
 
     struct HarfbuzzSpec;
@@ -39,7 +41,15 @@ mod inner {
 
         let mut sep = "cargo:include-path=";
 
-        dep.foreach_include_path(|p| {
+        let is_macos_external =
+            target_cfg!(target_os = "macos") && env::var("CARGO_FEATURE_EXTERNAL_HARFBUZZ").is_ok();
+
+        dep.foreach_include_path(|mut p| {
+            // HACK: On macOS, the default brew harfbuzz for some reason points into the harfbuzz
+            // folder itself, so we need to go up one level to the includes folder.
+            if is_macos_external {
+                p = p.parent().unwrap();
+            }
             print!("{}{}", sep, p.to_str().unwrap());
             sep = ";";
         });

--- a/crates/bridge_harfbuzz/build.rs
+++ b/crates/bridge_harfbuzz/build.rs
@@ -47,7 +47,7 @@ mod inner {
         dep.foreach_include_path(|p| {
             // HACK: On macOS, the default brew harfbuzz for some reason points into the harfbuzz
             // folder itself, so we need to go up one level to the includes folder.
-            if is_macos_external {
+            if is_macos_external && p.ends_with("harfbuzz") {
                 print!("{}{}", sep, p.parent().unwrap().to_str().unwrap());
                 sep = ";";
             }

--- a/crates/bridge_harfbuzz/build.rs
+++ b/crates/bridge_harfbuzz/build.rs
@@ -44,11 +44,12 @@ mod inner {
         let is_macos_external =
             target_cfg!(target_os = "macos") && env::var("CARGO_FEATURE_EXTERNAL_HARFBUZZ").is_ok();
 
-        dep.foreach_include_path(|mut p| {
+        dep.foreach_include_path(|p| {
             // HACK: On macOS, the default brew harfbuzz for some reason points into the harfbuzz
             // folder itself, so we need to go up one level to the includes folder.
             if is_macos_external {
-                p = p.parent().unwrap();
+                print!("{}{}", sep, p.parent().unwrap().to_str().unwrap());
+                sep = ";";
             }
             print!("{}{}", sep, p.to_str().unwrap());
             sep = ";";

